packages/captp/src/captp.js

```diff
@@ -311,29 +387,22 @@ export const makeCapTP = (
    */
   function convertSlotToVal(theirSlot, iface = undefined) {
     let val;
-    // Invert slot direction from other side.
-
-    // Inverted to prevent namespace collisions between slots we
-    // allocate and the ones the other side allocates.  If we allocate
-    // a slot, serialize it to the other side, and they send it back to
-    // us, we need to reference just our own slot, not one from their
-    // side.
-    const otherDir = theirSlot[1] === '+' ? '-' : '+';
-    const slot = `${theirSlot[0]}${otherDir}${theirSlot.slice(2)}`;
+    const slot = reverseSlot(theirSlot);
+
     if (!slotToVal.has(slot)) {
       // Make a new handled promise for the slot.
-      const pr = makeRemoteKit(slot);
+      const { promise, settler } = makeRemoteKit(slot);
       if (slot[0] === 'o' || slot[0] === 't') {
         if (iface === undefined) {
           iface = `Alleged: Presence ${ourId} ${slot}`;
         }
         // A new remote presence
         // Use Remotable rather than Far to make a remote from a presence
-        val = Remotable(iface, undefined, pr.resPres());
+        val = Remotable(iface, undefined, settler.resolvePresence());
```

Comment:
You declared the `resolvePresence` method as void return, but the implementation returns a presence that you seem to be using here intentionally.